lib/modules/datasource/rubygems/versions-endpoint-cache.ts

```diff
@@ -78,21 +73,28 @@ function parseFullBody(body: string): VersionsEndpointData {
   const contentLength = body.length;
   const contentTail = getContentTail(body);
 
-  return {
-    versionsEndpointSupported,
+  return Result.ok({
     packageVersions,
     syncedAt,
     contentLength,
     contentTail,
-  };
+  });
 }
 
-type VersionsEndpointResult =
-  | VersionsEndpointUnsupported
-  | VersionsEndpointData;
+type VersionsEndpointResult = Result<VersionsEndpointData, 'unsupported-api'>;
 
 export const memCache = new Map<string, VersionsEndpointResult>();
 
+function cacheResult(
+  registryUrl: string,
+  result: VersionsEndpointResult
+): void {
+  const registryHostname = parseUrl(registryUrl)?.hostname;
+  if (registryHostname === 'rubygems.org') {
```

Comment:
why do we only cache rubygems.org in memory?

Reply:
We don't want to cache private registries